### DISCUSSION TITLE
Some pieces are always on the bottom even when being grab.

### DISCUSSION THREAD

**Discussion by mania3908** [flbulgarelli/headbreaker#42] [Q&A] [answered]
Some pieces are always on the bottom even when being grab.

Is there a way to force the grab piece to the top so when the user grabs and drags a piece this go to the top

**Comment by flbulgarelli** [comment]
Hi @mania3908!  :wave: 

This does not appear to be an expected behaviour. Could you please provide a video and code sample?

**Reply by mania3908** [reply]
the user grabs a piece but this goes behind some of the pieces, i would want the piece im grabbing to be on top on everything while dragging. The code is from the examples.

![1](https://user-images.githubusercontent.com/7586960/210270669-634c1202-e766-41ab-b702-4d9bb39904bf.gif)

**Comment by flbulgarelli** [comment]
Oh, now I see you what you mean. 

I'd suggest setting the `z-index` of the piece's visual representation when you drag them. You can achieve this by accessing the native konva's `group`:

```javascript
const figures = Object.values(yourCanvasObjectHere.figures);
const figuresCount = figures.length;
figures.forEach(it => { 
  it.group.on('dragmove', () => { 
    it.group.zIndex(figuresCount) 
  });  
});
```

For example, now this is how it would look with the `soft` example in the docs: 

[Screencast from 03-01-23 12:52:56.webm](https://user-images.githubusercontent.com/677436/210393343-12c0ba0d-567a-4abf-9643-b6047dcd0ebc.webm)

I hope it helps!

**Comment by flbulgarelli** [comment, marked as answer]
I have just released a new version of `headbreaker` that ships with that functionality. You can check it here: https://flbulgarelli.github.io/headbreaker/#background
